contracts/ERC404.sol

```diff
@@ -118,53 +118,87 @@ abstract contract ERC404 is IERC404 {
   }
 
   function erc721TotalSupply() public view virtual returns (uint256) {
-    return _minted;
+    return minted;
   }
 
-  function erc721TokensBankedInQueue() public view virtual returns (uint256) {
+  function getERC721QueueLength() public view virtual returns (uint256) {
     return _storedERC721Ids.length();
   }
 
+  function getERC721TokensInQueue(
+    uint256 start_, 
+    uint256 count_
+  ) public view virtual returns (uint256[] memory) {
+    uint256[] memory tokensInQueue = new uint256[](count_);
+
+    for (uint256 i = start_; i < start_ + count_;) {
+      tokensInQueue[i - start_] = _storedERC721Ids.at(i);
+
+      unchecked {
+        ++i;
+      }
+    }
+
+    return tokensInQueue;
+  }
+
   /// @notice tokenURI must be implemented by child contract
   function tokenURI(uint256 id_) public view virtual returns (string memory);
 
   /// @notice Function for token approvals
   /// @dev This function assumes the operator is attempting to approve an ERC-721
-  ///      if valueOrId is less than the minted count. Note: Unlike setApprovalForAll,
+  ///      if valueOrId is less than the minted count. Unlike setApprovalForAll,
   ///      spender_ must be allowed to be 0x0 so that approval can be revoked.
+  ///
+  ///      Providing type(uint256).max for approval value results in an
+  ///      unlimited approval that is not deducted from on transfers.
   function approve(
     address spender_,
     uint256 valueOrId_
   ) public virtual returns (bool) {
     // The ERC-721 tokens are 1-indexed, so 0 is not a valid id and indicates that
     // operator is attempting to set the ERC-20 allowance to 0.
-    if (valueOrId_ <= _minted && valueOrId_ > 0) {
-      // Intention is to approve as ERC-721 token (id).
-      uint256 id = valueOrId_;
-      address erc721Owner = _getOwnerOf(id);
+    if (valueOrId_ <= minted && valueOrId_ > 0) {
```

Comment:
just an idea, not needed for this version: this logic could be a function since we reuse it everywhere, it's critical, and when we want to support burning it will need to be changed in many places.